include/flatbuffers/flatc.h

```diff
@@ -33,7 +33,8 @@ class FlatCompiler {
   struct Generator {
     typedef bool (*GenerateFn)(const flatbuffers::Parser &parser,
                                const std::string &path,
-                               const std::string &file_name);
+                               const std::string &file_name,
+                               const flatbuffers::IDLOptions& opts);
```

Comment:
Yes, `Parser` will extract the `known_attributes_` in constructor, but we need the `IDLOptions` for generator, it is a little strange to add a generating option to parser's known attributes. So, I prefer to pass the `IDLOptions` to Generator itself.

If you think use `Parser::known_attributes_` will be better, I can update the patch later.